### PR TITLE
ENH: move global to dict

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

File: xpdan/glbl.py [hunk 1]
```diff
@@ -61,11 +83,13 @@ def make_glbl(env_code=0, db=None):
...
         BASE_DIR = os.getcwd()
-        db = build_pymongo_backed_broker()
+        db1 = build_pymongo_backed_broker()
+        db2 = build_pymongo_backed_broker()
     else:
         # beamline
         BASE_DIR = os.path.abspath('/direct/XF28ID1/pe2_data')
-        from databroker.databroker import DataBroker as db
+        db1 = build_broker_from_config()
+        db2 = build_broker_from_config('an_metadatastore', 'an_filestore')
```
Comment (chiahaoliu) [inline xpdan/glbl.py]: I guess that is where #78 links to

Comment (CJ-Wright) [inline xpdan/glbl.py]: Yep

**Comment by CJ-Wright** [comment]
@sbillinge Ready for review.

File: xpdan/glbl.py [hunk 2]
```diff
@@ -108,46 +133,44 @@ def make_glbl(env_code=0, db=None):
...
+                scanplan_dir=SCANPLAN_DIR,
+                allfolders=ALL_FOLDERS,
+                archive_dir=USER_BACKUP_DIR,
+                owner=OWNER,
+                beamline_id=BEAMLINE_ID,
+                group=GROUP,
+                det_image_field=DET_IMAGE_FIELD,
+                dark_field_key=DARK_FIELD_KEY,
+                calib_config_name=CALIB_CONFIG_NAME,
+                exp_db=db1,
```
Comment (sbillinge) [inline xpdan/glbl.py]: wouldn't it be easier to follow if we called db1 as db_xptal and db2 as db_an or something like that?  Rather small point but shows intent.

File: xpdan/glbl.py [hunk 3]
```diff
@@ -108,46 +133,44 @@ def make_glbl(env_code=0, db=None):
...
+                an_db=db2,
+                mask_dict={'edge': 30, 'lower_thresh': 0.0,
```
Comment (sbillinge) [inline xpdan/glbl.py]: here we are hard-coding default numbers in glbl.py....is that a good idea?  wouldn't it be better for all numbers to be in the config?  Some of these numbers could change if we get a new detector, for example.

Comment (CJ-Wright) [inline xpdan/glbl.py]: You are quite right, we should stuff all of this into a configuration file.

File: xpdan/glbl.py (hunk 2 above)
Comment (CJ-Wright) [inline xpdan/glbl.py]: That is a good idea

**Comment by sbillinge** [comment]
@CJ-Wright I am ok for you to pull this in if you are ready when you have fixed those small things.  It will be good to get globals behind us....

**Comment by CJ-Wright** [comment]
Looking at your comments I think this has a ways to go.
- [ ] Rebase onto master so we can cover the tests.
- [ ] Change how the globals are made
   1. We should attempt to load the globals from a couple of sources similar to how MDS/FS are configured
   1. We should then try to post-process the resulting dictionary, creating the DB instances, temp files, and derived folders as needed
   1. Then we can all the globals to be loaded
   1. This will make us quite a bit more flexible as we can change many important configurations as needed without touching source code.

**Comment by CJ-Wright** [comment]
please see #87

**Pull request closed without merging**